srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java

```diff
@@ -143,7 +143,8 @@ public void discountBooks(Stream<Books> books) {
 	@After
 	public void setIsReviewable(CdsReadEventContext context, List<Books> books) {
 		String user = context.getUserInfo().getName();
-		List<String> bookIds = books.stream().map(b -> b.getId()).collect(Collectors.toList());
+		List<String> bookIds = books.stream().filter(b -> b.getId() != null).map(b -> b.getId())
+				.collect(Collectors.toList());
```

Comment:
Should we add a check if the `List` is empty?

Reply:
Yes, I think that would make sense. If `bookIds` is empty we can skip the fetching from the db and setting the field.